programs/vote/src/vote_state/mod.rs

```diff
@@ -36,9 +38,69 @@ pub const INITIAL_LOCKOUT: usize = 2;
 // Maximum number of credits history to keep around
 pub const MAX_EPOCH_CREDITS_HISTORY: usize = 64;
 
-// Offset of VoteState::prior_voters, for determining initialization status without deserialization
+// Offset of VoteState::pri : Clone + Debug {or_voters, for determining initialization status without deserialization
 const DEFAULT_PRIOR_VOTERS_OFFSET: usize = 82;
 
+// VoteTransactionClone hack is done so that we can derive clone on the tower that uses the
+// VoteTransaction trait object. Clone doesn't work here because it returns Self which is not
+// allowed for trait objects
+#[typetag::serde{tag = "type"}]
+pub trait VoteTransaction: VoteTransactionClone + Debug + Send {
```

Comment:
I would recommend using an enum instead of trait object to pass around two different kinds of votes

Reply:
I agree, I think at this point it's not worth rewriting but if I had to do it again i would use an enum.